app/uk/gov/hmrc/eoricommoncomponent/frontend/controllers/auth/AuthAction.scala

```diff
@@ -47,39 +47,55 @@ class AuthAction @Inject() (
 
   def ggAuthorisedUserWithEnrolmentsAction(requestProcessor: RequestProcessorSimple) =
     Action.async { implicit request =>
-      implicit val hc: HeaderCarrier =
-        HeaderCarrierConverter.fromHeadersAndSession(request.headers, Some(request.session))
+      authorise(requestProcessor)
+    }
 
-      authorised(AuthProviders(GovernmentGateway))
-        .retrieve(extendedRetrievals) {
-          case currentUserEmail ~ userCredentialRole ~ userAffinityGroup ~ userInternalId ~ userAllEnrolments ~ groupId =>
-            transformRequest(
-              Right(requestProcessor),
-              userAffinityGroup,
-              userInternalId,
-              userAllEnrolments,
-              currentUserEmail,
-              userCredentialRole,
-              groupId
-            )
-        } recover withAuthRecovery(request)
+  def ggAuthorisedUser(requestProcessor: RequestProcessorSimple) =
+    Action.async { implicit request =>
+      authorise(requestProcessor, checkPermittedAccess = false)
     }
 
+  private def authorise(requestProcessor: RequestProcessorSimple, checkPermittedAccess: Boolean = true)(implicit
+    request: Request[AnyContent]
+  ) = {
+    implicit val hc: HeaderCarrier =
+      HeaderCarrierConverter.fromHeadersAndSession(request.headers, Some(request.session))
+
+    authorised(AuthProviders(GovernmentGateway))
+      .retrieve(extendedRetrievals) {
+        case currentUserEmail ~ userCredentialRole ~ userAffinityGroup ~ userInternalId ~ userAllEnrolments ~ groupId =>
+          transformRequest(
+            Right(requestProcessor),
+            userAffinityGroup,
+            userInternalId,
+            userAllEnrolments,
+            currentUserEmail,
+            userCredentialRole,
+            groupId,
+            checkPermittedAccess
+          )
+      } recover withAuthRecovery(request)
+  }
+
   private def transformRequest(
     requestProcessor: Either[RequestProcessorExtended, RequestProcessorSimple],
     userAffinityGroup: Option[AffinityGroup],
     userInternalId: Option[String],
     userAllEnrolments: Enrolments,
     currentUserEmail: Option[String],
     userCredentialRole: Option[CredentialRole],
-    groupId: Option[String]
+    groupId: Option[String],
+    checkPermittedAccess: Boolean
   )(implicit request: Request[AnyContent]) = {
     val loggedInUser =
       LoggedInUserWithEnrolments(userAffinityGroup, userInternalId, userAllEnrolments, currentUserEmail, groupId)
```

Comment:
Just wondering if it's not better to create this loggedInUser val in authorise method. In that case this transform request method will have less parameters. The whole implementation will be cleaner probably.

Reply:
3 of the 5 params needed to create the user are used elsewhere in this method, so would still need to be passed.  So we'd take out 2 params from the method and add one (the user).  Not sure it's worth it.

Reply:
Yeah, that's true, but we don't need to pass them, because we can retrieve them from user object.

But I missed that before

Reply:
Good point!